irksome/descriptor.py

```diff
@@ -0,0 +1,30 @@
+class GalerkinDescriptor:
+    """
+    Base class for describing Galerkin-in-time methods in lieu of
+    a Butcher tableau.
+
+    :arg order: An integer indicating the order of the method
+    :kwarg basis_type: A string indicating the finite element family (either
+           `'Lagrange'` or `'Bernstein'`) or the Lagrange variant for the
+           test/trial spaces. Defaults to equispaced Lagrange elements.
+    :kwarg quadrature: A :class:`FIAT.QuadratureRule` indicating the quadrature
+            to be used in time, defaulting to GL with order points
+    """
+    def __init__(self, order, basis_type, quadrature):
+        self.order = order
+        self.basis_type = basis_type
```

Comment:
For greater generality, we could also allow separate types for test and trial bases?

Reply:
This feels like a thing we should allow, even if it defaults to being the same